src/box/schema.cc

```diff
@@ -487,3 +489,307 @@ schema_find_name(enum schema_object_type type, uint32_t object_id)
 	return NULL;
 }
 
+/**
+ * Drop the constraint or foreign key. If fieldno is equal to UINT32_MAX then
+ * the tuple constraint or tuple foreign key is dropped, otherwise the field
+ * constraint or field foreign key is dropped.
+ */
+int
+constraint_drop(uint32_t space_id, const char *name, uint32_t name_len,
```

Comment:
Having this new API in schema.h looks odd. Mostly because this file had nothing to do with the storage layer before (and this is why it didn't need box.h). All was in box.h/.cc like `box_insert()`, `box_sequence_next()`, `boxk()` and all the similar functions. I would consider moving it all there. Up to you.

Reply:
Thanks, moved.